### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by AndrewS-123**
Added leader and follower motors  and encoder to subsystem. Added default run methods to the subsystem. Added RunIntake command, runs motors at a constant speed. added timeout and line break sensor as end condition for RunIntake, added intake config, added unused PID control.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 1]
```diff
@@ -1,18 +1,160 @@
...
-  public Intake() {}
+  private final CANSparkMax m_rollerFollowerMotor; // left motor (Rollers)
+  private final CANSparkMax m_rollerLeaderMotor; // right motor (Rollers)
+
+  private final SparkPIDController m_rollerPidController; // Roller velocity PID controllers
+  private final RelativeEncoder m_rollerEncoder; // Roller Relative Encoder
+
+  private final DigitalInput m_intakeSensor; // Line break note sensor
+
+  private boolean m_testModeCheck1; // Booleans for if test mode is enabled
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: remove this, we don't need PID constants checks (or PID at all)

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 2]
```diff
@@ -1,18 +1,160 @@
...
+    m_rollerPidController = m_rollerLeaderMotor.getPIDController();
+    m_rollerEncoder = m_rollerLeaderMotor.getEncoder();
+    zeroEncoders();
+
+    m_intakeSensor = new DigitalInput(RobotConstants.IntakeConfig.kLineBreakSensor);
+
+    // set Roller PID coefficients
+    m_rollerPidController.setP(RobotConfig.IntakeConfig.kRollerP);
+    m_rollerPidController.setI(RobotConfig.IntakeConfig.kRollerI);
+    m_rollerPidController.setD(RobotConfig.IntakeConfig.kRollerD);
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: we don't need PID, remove

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 3]
```diff
@@ -1,18 +1,160 @@
...
+        RobotConfig.IntakeConfig.kRollerDGainKey, RobotConfig.IntakeConfig.kRollerD);
+    SmartDashboard.putNumber(
+        RobotConfig.IntakeConfig.kRollerFFGainKey, RobotConfig.IntakeConfig.kRollerFF);
+    SmartDashboard.putNumber(
+        RobotConfig.IntakeConfig.kRollerIZoneKey, RobotConfig.IntakeConfig.kRollerIZone);
+    SmartDashboard.putNumber(
+        RobotConfig.IntakeConfig.kRollerMinOutputKey, RobotConfig.IntakeConfig.kRollerMinOutput);
+    SmartDashboard.putNumber(
+        RobotConfig.IntakeConfig.kRollerMaxOutputKey, RobotConfig.IntakeConfig.kRollerMaxOutput);
+    SmartDashboard.putBoolean(RobotConfig.IntakeConfig.kTestCheck1Key, m_testModeCheck1);
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: remove these gains

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 4]
```diff
@@ -4,4 +4,34 @@
...
+    public static final double kRollerP = 0;
+    public static final double kRollerI = 0;
+    public static final double kRollerD = 0;
+    public static final double kRollerFF = 0;
+    public static final double kRollerIZone = 0;
+    public static final double kRollerMinOutput = 0;
+    public static final double kRollerMaxOutput = 0;
+
+    // Smart dashboard roller keys
+    public static final String kRollerPGainKey = "Roller P Gain";
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/constants/RobotConfig.java]: remove gains and keys

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    m_intake = subsystem;
+    m_timer = new Timer();
+
+    addRequirements(subsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_timer.start();
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/commands/intake/RunIntake.java]: Why a timer?

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  @Override
+  public void end(boolean interrupted) {
+    m_intake.stop();
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return (m_intake.isIndexed()
+        || m_timer.advanceIfElapsed(RobotConfig.IntakeConfig.kRunIntakeTimeoutTime));
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/commands/intake/RunIntake.java]: Why? We can just use a timed command?

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 7]
```diff
@@ -4,4 +4,34 @@
  * Software config settings (e.g. max speed, PID values). For hardware constants @see
  * RobotConstants"
  */
-public class RobotConfig {}
+public class RobotConfig {
+  public static final class IntakeConfig {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConfig.java]: Why do we need this?

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 8]
```diff
@@ -4,4 +4,15 @@
  * Software/hardware constants (e.g. CAN IDs, gear ratios, field measurements, etc.). For software
  * configs @see RobotConfig
  */
-public final class RobotConstants {}
+public final class RobotConstants {
+  public static final class IntakeConfig {
+    // Placeholder CAN IDs, replace later
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConstants.java]: Mark with TODO comments

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 9]
```diff
@@ -1,18 +1,160 @@
...
+    SmartDashboard.putBoolean(RobotConfig.IntakeConfig.kTestCheck2Key, m_testModeCheck2);
+  }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
+
+    boolean testCheck1 = SmartDashboard.getBoolean(RobotConfig.IntakeConfig.kTestCheck1Key, false);
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Why is this needed?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 10]
```diff
@@ -1,18 +1,160 @@
...
+    boolean testCheck2 = SmartDashboard.getBoolean(RobotConfig.IntakeConfig.kTestCheck2Key, false);
+
+    // Checks if test mode is enabled
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: What's the purpose of these?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 11]
```diff
@@ -1,18 +1,160 @@
...
+      // read PID coefficients from SmartDashboard
+      double p = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerPGainKey, 0);
+      double i = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerIGainKey, 0);
+      double d = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerDGainKey, 0);
+      double iz = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerIZoneKey, 0);
+      double ff = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerFFGainKey, 0);
+      double max = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerMaxOutputKey, 0);
+      double min = SmartDashboard.getNumber(RobotConfig.IntakeConfig.kRollerMinOutputKey, 0);
+
+      // checks PID values against Smartdash board
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: We don't need these right?

**Review by garrettwitowski — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 10 above)
Comment (EmDash00) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: I recommended them for test mode for PID constant setting so they aren't accidentally loaded or something, but I realized `testPeriodic()` might be a better function for them. Either way, since we aren't doing PID on the intake, they are unnecessary.

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 11 above)
Comment (EmDash00) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: No we don't.

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 1 above)
Comment (EmDash00) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Yep.

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 12]
```diff
@@ -1,18 +1,160 @@
...
+  public void run(boolean isReversed) {
+    double runSpeed =
+        isReversed
+            ? -RobotConfig.IntakeConfig.kDefaultSpeed
+            : RobotConfig.IntakeConfig.kDefaultSpeed;
+    run(runSpeed);
+  }
+
+  // Checks if note is indexed
+  public boolean isIndexed() {
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Remove this. We're making the check part of shooter.

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 13]
```diff
@@ -1,18 +1,160 @@
...
+    return m_intakeSensor.get();
+  }
+
+  // Stops the motor
+  public void stop() {
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java:31]: Make this a javadoc docstring.

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 14]
```diff
@@ -1,18 +1,160 @@
...
+    m_rollerLeaderMotor.stopMotor();
+  }
+
+  // Zeros the encoder(s)
+  public void zeroEncoders() {
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Unnecessary function for intake. Remove.

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 15]
```diff
@@ -1,18 +1,160 @@
...
+  }
+
+  /**
+   * Runs the intake at a default speed The motor can be reversed by inputing a true as its
+   * parameter
+   *
+   * @param isReversed The input to reverse the motor speed
+   */
+  public void run(boolean isReversed) {
+    double runSpeed =
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Use of ternary operator is discouraged when it reduces readability (like here). Use an if statement for better readability.

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 16]
```diff
@@ -1,18 +1,160 @@
...
 
-  @Override
-  public void simulationPeriodic() {
-    // This method will be called once per scheduler run during simulation
+  /**
+   * Sets the target rpm
+   *
+   * @param targetRPM Target rpm
+   */
+  public void setTargetRPM(double targetRPM) {
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: We're not using PID on the intake. Remove.

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 7 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java]: Intake config formerly held PID constants. After PID constants were removed, all that's stored in there is Default Speed and Timeout time

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 4 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java]: Done in the "Removed PID" commit

File: src/main/java/frc/robot/constants/RobotConstants.java (hunk 8 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConstants.java]: Done in the PID + update commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 1 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed alongside PID in "Removed PID commit"

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 2 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed PID

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 3 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Done in PID remove

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 11 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed in "Removed PID" PID

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 6 above)
Comment (Irishumanoid) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: I don't see any problems with setting a timeout internally, it does the same thing.

File: src/main/java/frc/robot/constants/RobotConfig.java
Comment (Irishumanoid) [inline src/main/java/frc/robot/constants/RobotConfig.java:1]: There are some merge conflicts (different versions of the same code) between this branch and develop. Resolve those by pulling develop into your branch and using vim or the vscode editor to choose which version you want to keep (or a combination of both).

Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java:1]: We have pulled and resolved conflicts

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 10 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: They have been removed

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 6 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: We are going with a timer

**Review by EmDash00 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 5 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: Removed in commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 15 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: This method was removed, along with its ternary operator in commit removing run(isReversed)

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 12 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Done in "Removed isIndexed" commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 13 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java:31]: Converted to a javadoc docstring in "Changed stop method" commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 14 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed in "removed zeroEncoders" commit

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 6 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: > We are going with a timer

Change, we've remove it in commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 16 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: removed in PID commit

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+   */
+  public RunIntake(Intake subsystem) {
+    m_intake = subsystem;
+
+    addRequirements(subsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/commands/intake/RunIntake.java:24]: intake should be run in execute

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 18]
```diff
@@ -1,18 +1,43 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.RobotConstants.IntakeConstants;
 
 public class Intake extends SubsystemBase {
-  /** Creates a new ExampleSubsystem. */
-  public Intake() {}
+  private final CANSparkMax m_rollerMotor; // right motor (Rollers)
+
+  private final RelativeEncoder m_rollerEncoder; // Roller Relative Encoder
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: don't need extra lines, when you need to get position or velocity you can just run motor.getEncoder() etc and your code will be more compact

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 19]
```diff
@@ -1,18 +1,43 @@
...
+  public Intake() {
+    m_rollerMotor = new CANSparkMax(IntakeConstants.kMotorID, MotorType.kBrushless);
+
+    m_rollerEncoder = m_rollerMotor.getEncoder();
+  }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
+    SmartDashboard.putNumber("intake/roller velocity", m_rollerEncoder.getVelocity());
```
Comment (EmDash00) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: Remove. Unnecessary for intake.

**Review by EmDash00 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 19 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed accidentally in "Remove encoder" commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 18 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java]: Removed encoder in "removed encoder" commit

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 17 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java:24]: Moved to execute in "moved run to execute" commit

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.RobotConfig;
+import frc.robot.subsystems.intake.Intake;
+
+public class RunIntake extends Command {
+  private final Intake m_intake;
+
+  /**
+   * Creates a new RunIntake command.
```
Comment (EmDash00) [inline src/main/java/frc/robot/commands/intake/RunIntake.java]: Which does what?

Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: Updated docstring to include what exactly that does in "Updated runintake docstring" commit

**Review by EmDash00 — APPROVED** [review]

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+public class RunIntake extends Command {
+  private final Intake m_intake;
+
+  /**
+   * Creates a new RunIntake command, which runs the roller motor on the intake subsystem to intake
+   * a note
+   *
+   * @param subsystem The subsystem used by this command.
+   */
+  public RunIntake(Intake subsystem) {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/commands/intake/RunIntake.java]: nit: we should bias towards using descriptive names for variables. Something like
```
public RunIntake(Intake intake) { ...
```

would be more clear.

File: src/main/java/frc/robot/commands/intake/RunIntake.java [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_intake.run(RobotConfig.IntakeConfig.kDefaultSpeed);
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/commands/intake/RunIntake.java]: nit: this should probably be called from `initialize()`, rather than from `execute()` because `execute()` is intended for periodic tasks

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 23]
```diff
@@ -25,4 +25,12 @@ public static class DriveConfig {
     public static final double MIN_ANGLE_SLEW_RATE = 0.45;
     public static final double MAX_ANGLE_SLEW_RATE = 0.85;
   }
+
+  public static final class IntakeConfig {
+    // TODO: update intake motors default spped
+    public static final double kDefaultSpeed = 0;
+
+    // TODO: update timeout time (in seconds)
+    public static final double kRunIntakeTimeoutTime = 0;
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConfig.java]: nit: TimeoutTime is redundant -- `kRunIntakeTimeout` is sufficient, or else something like `kRunIntakeTimeoutSecs` (if you want to be more descriptive).

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 24]
```diff
@@ -25,4 +25,12 @@ public static class DriveConfig {
     public static final double MIN_ANGLE_SLEW_RATE = 0.45;
     public static final double MAX_ANGLE_SLEW_RATE = 0.85;
   }
+
+  public static final class IntakeConfig {
+    // TODO: update intake motors default spped
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConfig.java]: nit: typo :)

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 25]
```diff
@@ -25,4 +25,12 @@ public static class DriveConfig {
...
+    public static final double kDefaultSpeed = 0;
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConfig.java:32]: nit: ideally we should state what units this is in

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 26]
```diff
@@ -114,4 +114,13 @@ public static final class SwerveModuleConstants {
       public static final int kTurningMotorCurrentLimit = 20; // amps
     }
   }
+
+  public static final class IntakeConstants {
+
+    // TODO: update placeholder digital input ID
+    public static final int kLineBreakSensor = -1;
+
+    // TODO: update placerholder roller Motor ID
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/constants/RobotConstants.java]: nit: typo

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 27]
```diff
@@ -1,18 +1,34 @@
...
 
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.RobotConstants.IntakeConstants;
 
 public class Intake extends SubsystemBase {
-  /** Creates a new ExampleSubsystem. */
-  public Intake() {}
+  private final CANSparkMax m_rollerMotor; // Intake roller motor
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/intake/Intake.java:9]: nit: prefer 2 spaces in between code and comments on the same line, i.e.
```
private final CANSparkMax m_rollerMotor;  // Intake roller motor
```

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 28]
```diff
@@ -1,18 +1,34 @@
...
+
+  // Constructs intake and initializes motor
+  public Intake() {
+    m_rollerMotor = new CANSparkMax(IntakeConstants.kMotorID, MotorType.kBrushless);
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/intake/Intake.java:13]: Just a note: depending on how the motor is mounted you may need to set its "inverted" state to `true`. No action necessary at this time.

**Review by garrettwitowski — COMMENTED** [review]
Overall looks good, just a bunch of nits but you can address those in a subsequent revision (i.e., after we merge this).

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 21 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: Changed name in commit

File: src/main/java/frc/robot/commands/intake/RunIntake.java (hunk 22 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/commands/intake/RunIntake.java]: Done in commit

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 24 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java]: Fixed in commit

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 25 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java:32]: Done in commit

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 23 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConfig.java]: Done in commit

File: src/main/java/frc/robot/constants/RobotConstants.java (hunk 26 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/constants/RobotConstants.java]: Fixed in commit

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 27 above)
Comment (ChainedGCH) [reply src/main/java/frc/robot/subsystems/intake/Intake.java:9]: That is in conflict with spotless

Comment (garrettwitowski) [reply src/main/java/frc/robot/subsystems/intake/Intake.java:9]: OK, good to know. Thanks. We use a different style guide at work.

**Review by garrettwitowski — APPROVED** [review]
Looks good, thanks for addressing everything!
